opal/resource.go

```diff
@@ -428,6 +443,79 @@ func resourceResourceRead(ctx context.Context, d *schema.ResourceData, m any) di
 			remoteInfoIList = append(remoteInfoIList, map[string]any{
 				"aws_permission_set": permissionSetIList,
 			})
+		case opal.RESOURCETYPEENUM_AWS_IAM_ROLE:
+			// TODO: Either make metadata and remote_info match or create a new object for this
+			var metadata map[string]opal.ResourceRemoteInfoAwsIamRole
+			if err := json.Unmarshal([]byte(*resource.Metadata), &metadata); err != nil {
+				return diagFromErr(ctx, err)
+			}
+			if awsRole, ok := metadata["aws_role"]; ok {
+				roleIList := make([]any, 0, 1)
+				roleIList = append(roleIList, map[string]any{
+					"arn":        awsRole.Arn,
+					"account_id": awsRole.AccountId,
+				})
+				remoteInfoIList = append(remoteInfoIList, map[string]any{
+					"aws_iam_role": roleIList,
+				})
+			} else if !ok {
+				return diagFromErr(ctx, errors.New("resource metadata is missing aws_role"))
+			}
+
+		case opal.RESOURCETYPEENUM_AWS_EC2_INSTANCE:
+			var metadata opal.ResourceRemoteInfo
+			if err := json.Unmarshal([]byte(*resource.Metadata), &metadata); err != nil {
+				return diagFromErr(ctx, err)
+			}
+			if metadata.AwsEc2Instance == nil {
+				return diagFromErr(ctx, errors.New("resource metadata is missing aws_ec2_instance"))
+			}
+			instanceIList := make([]any, 0, 1)
+			instanceIList = append(instanceIList, map[string]any{
+				"instance_id": metadata.AwsEc2Instance.InstanceId,
+				"region":      metadata.AwsEc2Instance.Region,
+				"account_id":  metadata.AwsEc2Instance.AccountId,
+			})
+			remoteInfoIList = append(remoteInfoIList, map[string]any{
+				"aws_ec2_instance": instanceIList,
+			})
+		case opal.RESOURCETYPEENUM_AWS_RDS_MYSQL_INSTANCE,
+			opal.RESOURCETYPEENUM_AWS_RDS_POSTGRES_INSTANCE:
+			var metadata opal.ResourceRemoteInfo
+			if err := json.Unmarshal([]byte(*resource.Metadata), &metadata); err != nil {
+				return diagFromErr(ctx, err)
+			}
+			if metadata.AwsRdsInstance == nil {
+				return diagFromErr(ctx, errors.New("resource metadata is missing aws_rds_instance"))
+			}
+			databaseIList := make([]any, 0, 1)
+			databaseIList = append(databaseIList, map[string]any{
+				"instance_id": metadata.AwsRdsInstance.InstanceId,
+				"region":      metadata.AwsRdsInstance.Region,
+				"account_id":  metadata.AwsRdsInstance.AccountId,
```

Comment:
looks like RDS instances are also missing `AccountId` in the metadata?

https://github.com/opalsecurity/opal/blob/7447166372367f6b18faa381f44e988e24b79056/web/backend/models/schema/metadata.go#L76-L81